Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
       PEP written by Marc-André Lemburg.
 
 
+.. We use the following practises for SQL code:
+   - UPPERCASE for keywords
+   - snake_case for schema
+   - single quotes for string literals
+   - singular for table names
+   - if needed, use double quotes for table and column names
+
 .. _sqlite3-tutorial:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating a database of Monty Python movies.
+It assumes a fundamental understanding of database concepts,
+including `cursors`_ and `transactions`_.
+
+First, you'll need to create a new database to hold the movie data
+and open a database connection to allow :mod:`!sqlite3` to work with it.
+Call the :func:`sqlite3.connect` function to
+to create a connection to a database :file:`tutorial.db`
+in the current working directory,
+implicitly creating it if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+you'll use a database cursor.
```

Comment:
```suggestion
you need to use a database cursor.
```
I prefer this than "you'll".  The latter sounds a bit demanding, since it sounds like the tutorial is telling the user that they must do what the tutorial says.  "You need to use" sounds less forceful to me, simply informing the user what they need to do to proceed to the next step.

If this change is accepted, there are more instances that need to be updated.

Reply:
IMO,  at least as a native American-English speaker, the suggestion actually sounds slightly _more_ demanding to me, whereas "you'll" doesn't as much—but obviously it varies from person to person, so take that as just another data point. On the other hand I tend to avoid contractions in documentation, since it can be harder to understand for some international audiences and many doc style guides recommend against it.

Reply:
IMO,  at least as a native American-English speaker, the suggestion actually sounds slightly _more_ demanding to me, whereas "you'll" doesn't as much—but obviously it varies from person to person, so take that as just another data point. On the other hand I tend to avoid contractions in documentation, since it can be harder to understand for some international audiences and many doc style guides recommend against it.

Reply:
To elaborate a bit, this is how different options sound to me:
* "you'll use a db cursor" sounds like an inexorable prediction that not only you will do something, but that you will do it that way.
* "you must use a db cursor" sounds like an order (which I could possible decide to ignore and deal with the consequences).
* "you have to use a db cursor" sounds a bit more matter-of-factly, without necessarily implying that the user will do it (only if they want to execute queries)..
* "you need to use a db cursor" similar to "have to", but slightly less forceful IMHO (even though admittedly it depends on the context and tone).
* "you can use a db cursor" sounds like there are other options too, but you can pick this one.

Reply:
FWIW, as a non-native speaker, here's how I perceive this, ordered by "niceness", most stringent last. I exclude "you can use", since it does not make sense when describing unambiguous steps.

- you'll use / you will use
- you need to use
- you must use
- you have to use

I don't know what's best regarding "you will ned to use" and "you need to use"

Reply:
I do agree with CAM regarding contractions; for now, I'll change it to "you will use".

Reply:
> I don't know what's best regarding "you will ned to use" and "you need to use"

They say different things; the former introduces what _will_ be required, while "you need to use" is a command. In this case, the former is what you want. It is also much nicer. However, "you need to use" is nicer and more appropriate, IMO, than "you will use" and avoids some of the potential concern @ezio-melotti had about telling students what they _will_ inevitably do in the future, as opposed to what they will _need_ to do (if they are following the tutorial).

> I do agree with CAM regarding contractions; for now, I'll change it to "you will use".

IMO, to me "you will use" is a fair bit less "nice" and friendly than "you'll use", so we might need to decide which is the priority here. But "you will need to use" is about equally nice and more appropriate as a replacement.

Reply:
BTW; isn't "you will need to use" slightly more polite than "you need to use"?

Reply:
As a non-native speaker, I trust your judgements here.

Reply:
> BTW; isn't "you will need to use" slightly more polite than "you need to use"?

Not just more polite, but as I mention also more appropriate in this particular context, and helps address @ezio-melotti 's concern:

> They say different things; the former introduces what will be required, while "you need to use" is a command. In this case, the former is what you want. It is also much nicer. However, "you need to use" is nicer and more appropriate, IMO, than "you will use" and avoids some of the potential concern @ezio-melotti had about telling students what they will inevitably do in the future, as opposed to what they will need to do (if they are following the tutorial).